utils/src/lib.rs

```diff
@@ -0,0 +1,13 @@
+// Copyright 2019-2024 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! Subxt utils.
+
+// Internal helper macros
+#[macro_use]
+mod macros;
+
+macros::cfg_fetch_metadata! {
```

Comment:
I want to add `fetch_chain_spec` here as well but I want to see whether James is happy with this and I'll do in another PR